test/bridge/testase.py

```diff
@@ -0,0 +1,261 @@
+        assert np.allclose(ase_data.atomcoords, [data.atomcoords[-1]])
+        assert np.allclose(ase_data.atomnos, data.atomnos)
+        assert np.allclose(
+            ase_data.atomcharges["mulliken"], data.atomcharges["mulliken"]
+        )
+        assert np.allclose(
+            ase_data.atomspins["mulliken"], data.atomspins["mulliken"]
+        )
+        assert np.allclose(ase_data.atommasses, data.atommasses)
+        assert np.isclose(ase_data.charge, data.charge)
+        assert np.isclose(ase_data.mult, data.mult)
+        assert np.isclose(ase_data.natom, len(data.atomnos))
+        assert np.isclose(ase_data.temperature, 0)
+
+        # make sure our object is compatible with ASE API
+        dvb_sp_un.calc = EMT(label="dvb_sp_un")  # not a serious calculator!
+
+        # converting back should give updated results
+        ase_data = cclib2ase.makecclib(dvb_sp_un)
+        assert np.allclose(ase_data.atomcoords, [data.atomcoords[-1]])
+        assert np.allclose(ase_data.atomnos, data.atomnos)
+        assert np.allclose(ase_data.atommasses, data.atommasses)
+        assert np.allclose(
+            ase_data.atomcharges["mulliken"], data.atomcharges["mulliken"]
+        )
+        assert np.allclose(
+            ase_data.atomspins["mulliken"], 0
+        )  # spin densities not supported but overwritten by EMT
+        assert np.isclose(ase_data.charge, data.charge)
+        assert np.isclose(ase_data.mult, 1)
+        assert np.isclose(ase_data.natom, len(data.atomnos))
+        assert np.isclose(ase_data.temperature, 0)
+        assert np.allclose(ase_data.scfenergies, [7.016800805424298])
+        assert np.allclose(
+            ase_data.grads,
+            [
+                [
+                    [0.00601771, -0.00030069, -0.0],
+                    [-0.00601771, 0.00030069, -0.0],
+                    [0.03929838, -0.06541943, -0.0],
+                    [-0.03929838, 0.06541943, -0.0],
+                    [-0.03986435, -0.06680547, -0.0],
+                    [0.03986435, 0.06680547, -0.0],
+                    [-0.02627055, 0.04620284, -0.0],
+                    [0.02627055, -0.04620284, -0.0],
+                    [0.02699466, 0.04770784, -0.0],
+                    [-0.02699466, -0.04770784, -0.0],
+                    [-0.01551952, 0.07550522, -0.0],
+                    [0.01551952, -0.07550522, -0.0],
+                    [0.02016697, -0.04821387, -0.0],
+                    [-0.02016697, 0.04821387, -0.0],
+                    [-0.05412095, -0.06691358, -0.0],
+                    [0.05412095, 0.06691358, -0.0],
+                    [-0.01446204, 0.04423546, -0.0],
+                    [0.01446204, -0.04423546, -0.0],
+                    [0.04468942, -0.00502891, -0.0],
+                    [-0.04468942, 0.00502891, -0.0],
+                ]
+            ],
+        )
+        assert (
+            np.abs(dvb_sp_un.get_forces()).sum() > np.abs(ase_data.grads).sum()
```

Comment:
I'm not sure this check is meaningful, since on my machine it's comparing 2.4424906541753444e-15 to -1.5612511283791264e-17 for a quantity itself (EMT gradient) that isn't meaningful for the system. I'd replace both this assert and the full one above with a check for the gradient existing with the correct shape, and *not* being present on the original `data`.

Reply:
Right, done. Let me know if what I did is what you've expected (I tested for `AttributeError` in the case of the absent `grads` property).